src/rules.ts

```diff
@@ -3,33 +3,46 @@ import {DefaultAzureCredential} from '@azure/identity';
 import {AzureClient} from './azure';
 import {ScanResult} from './scanner';
 
-export type Rule = IResourceGraphRule | IDummyRule;
+export type RuleContext = ResourceGraphRuleContext | DummyRuleContext;
 
-interface BaseRule {
+export interface ResourceGraphRuleContext {
+  type: 'resourceGraph';
```

Comment:
I'm a little fuzzy on this. It seems that rules with a type property can no longer exist independently of the concrete descriptor of the thing that we want to run them against? Seems like a step backwards. It looks like we have no longer have a discriminator on our rules, so we have to go inspect properties of the rule objects to determine what it is